servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpServerBuilder.java

```diff
@@ -163,6 +163,15 @@ public final HttpServerBuilder disableDrainingRequestPayloadBody() {
         return this;
     }
 
+    /**
+     * Set if the <a href="https://tools.ietf.org/html/rfc7230#section-4.4">Trailer</a> header is required to process
+     * trailers.
+     * @param requireTrailerHeader {@code true} if <a href="https://tools.ietf.org/html/rfc7230#section-4.4">Trailer</a>
+     * header is required to accept trailers. {@code false} assumes trailers may be present if other criteria allows.
+     * @return {@code this}
+     */
+    public abstract HttpServerBuilder requireTrailerHeader(boolean requireTrailerHeader);
```

Comment:
The default value respects `Trailer` header as optional. This option offers an optimization to avoid more costly intermediate trailers preservation operations if they aren't explicitly specified as being used (basically OK to drop trailers if this is true and the request/response doesn't indicate trailers are present).

Reply:
The most frequent use-case for trailers is grpc, which doesn't use `Trailer` header. HTTP/2 supports it but doesn't suggest or require using it if the trailing `HEADERS` frame is expected.

Other use-cases for trailers are rare and it's not always the case that users use `Trailer` header. I'm just not sure if anyone will find this method useful.
If the performance is the main concern, we can try to run the benchmarks with and without this flag to see the impact. If it's negligible for the use-case when there are no trailers (most of HTTP/1.1 users with streaming API), I would propose to defer adding it to the public API (remove or make pkg-private).

Reply:
Discussed offline, this is only for receiving side (`newTransport[Request|Response]`). Let's give it a try and perf-test with gRPC (server can use it because [gRPC requests](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests) never have trailers).

Agreed with the offline proposal to clarify that in javadoc.

Also, consider not using `boolean` param because we don't need to flip it back and forth. The proposed `allowIgnoreTrailersIfNoHeader()` or just `ignoreTrailersIfNoHeader()` sgtm.

Reply:
I updated the method name and clarified the javadocs.